src/RealMeService.php

```diff
@@ -827,11 +851,14 @@ private function getRequestedAuthnContext()
      */
     public function getAuth()
     {
-        if(isset($this->auth)) return $this->auth;
+        if (isset($this->auth)) {
+            return $this->auth;
+        }
 
         // If we're behind a trusted proxy, force onelogin to use the HTTP_X_FORWARDED_FOR headers to determine
         // protocol, host and port
-        if(TRUSTED_PROXY) {
+        // @todo address this issue
+        if (false) { // if TRUSTED_PROXY
```

Comment:
What's the issue at the moment?

I don't know anything about this, but having had a quick good it looks like TRUSTED_PROXY isn't a SilverStripe specific constant, and `OneLogin_Saml2_Utils::setProxyVars(true)` will mean it internally handles the logic around the forwarded headers

Reply:
Ah, I forgot about this... `TRUSTED_PROXY` was a constant set by SS3, now it's been removed... https://github.com/silverstripe/silverstripe-framework/pull/7052 - specifically https://github.com/silverstripe/silverstripe-framework/pull/7052/commits/ccc86306b6a0d6eb9bbb8b41b209b8595393d591 . I figured the correct way to do this was to set the SAML host, port and protocol manually but I didn't get around to it. If you have any other suggestions I'll welcome them.

Reply:
That won't work. The middleware sets the IP of the request to the forwarded IP after it validates a trusted proxy. This means that subsequent calls to `isTrustedProxy` will be checking a different IP address (perhaps one that isn't whitelisted)

Reply:
Ah ok. You seem like you're more in the headspace with it right now so I'll leave it to you =)